playbooks/deploy.yml

```diff
@@ -9,64 +9,100 @@
+            ) | bool 
+          }}"
+        
+        snapshots_repository: "{{ ota_apps_config['@opentermsarchive/engine'].recorder.snapshots.storage.git.repository is defined and ota_apps_config['@opentermsarchive/engine'].recorder.snapshots.storage.git.repository }}"
+        snapshots_path: "{{ ota_apps_config['@opentermsarchive/engine'].recorder.snapshots.storage.git.path is defined and ota_apps_config['@opentermsarchive/engine'].recorder.snapshots.storage.git.path }}"
+        
+        versions_repository: "{{ ota_apps_config['@opentermsarchive/engine'].recorder.versions.storage.git.repository is defined and ota_apps_config['@opentermsarchive/engine'].recorder.versions.storage.git.repository }}"
+        versions_path: "{{ ota_apps_config['@opentermsarchive/engine'].recorder.versions.storage.git.path is defined and ota_apps_config['@opentermsarchive/engine'].recorder.versions.storage.git.path }}"
+        
+        collection_api_basePath: "{{ ota_apps_config['@opentermsarchive/engine']['collection-api'].basePath is defined and ota_apps_config['@opentermsarchive/engine']['collection-api'].basePath }}"
+        collection_api_port: "{{ ota_apps_config['@opentermsarchive/engine']['collection-api'].port is defined and ota_apps_config['@opentermsarchive/engine']['collection-api'].port }}"
+        
+        federation_api_basePath: "{{ ota_apps_config['@opentermsarchive/federation-api'].basePath is defined and ota_apps_config['@opentermsarchive/federation-api'].basePath }}"
+        federation_api_port: "{{ ota_apps_config['@opentermsarchive/federation-api'].port is defined and ota_apps_config['@opentermsarchive/federation-api'].port }}"
+    
+    - name: Install infrastructure
+      become: true
+      tags: [infrastructure]
+      block:
+        - name: Install Node
+          ansible.builtin.include_role:
+            name: node
+        
+        - name: Install PM2
+          ansible.builtin.include_role:
+            name: pm2/install
+
+        - name: Install Chromium
+          ansible.builtin.include_role:
+            name: chromium
+          when: chromium_required
+
+        - name: Install Nginx 
+          ansible.builtin.include_role:
+            name: nginx/install
       
-      - name: Install PM2
-        ansible.builtin.include_role:
-          name: pm2
-        vars:
-          ota_pm2_install_only: true
+        - name: Install Mongo
+          ansible.builtin.include_role:
+            name: mongo/install
+          when:
+            - mongo_required
 
-      - name: Install Chromium
-        ansible.builtin.include_role:
-          name: chromium
-        when: ota_apps_config['@opentermsarchive/engine'] is defined
+    - name: Configure Mongo
+      ansible.builtin.include_role:
+        name: mongo/configure
+        apply:
+          become: true
```

Comment:
I decided to manage all `become: true` settings at the playbook level for the following reasons:
- Centralization of privilege escalation in one place.
- Setting `become: true` once at the playbook level simplifies the roles, avoiding repetitive configuration in each task.

However, this approach has some drawbacks:
- It might escalate privileges for tasks that don't require it, but this it seems to me that it is not an issue in our case as each task in infrastructure need privileges.
- It reduces the control over which specific tasks need privilege escalation. However, the way our roles are structured means this granularity is not necessary.

Another solution considered was using `block` in roles to wrap all tasks with a `become: true` instruction. However, similar to defining global variables, I believe it is clearer to specify where privilege escalation will occur at the entry point of the playbook.